### DISCUSSION TITLE
What does the name mean?

### DISCUSSION THREAD

**Discussion by Nikoichu** [florisboard/florisboard#1735] [General]
I'm just curious, how did you come up with the name "Florisboard" and if there is some symbolic meaning you want to convey with the name, especially the "Floris-" part.

**Comment by patrickgold** [comment]
There's no special meaning behind it (at least none I am aware of). At the start of the project I was on the search for a good name that a) wasn't already in use and b) is recognizable and not some super generic one. Still I wanted to have "board" from "keyboard" in it, to immediately tell it is probably a keyboard.

So I searched for names in my head until I came up with some names. Among those was also "FlorisBoard", a name which I immediately liked and after some research I also chose to use. @Glitchy-Tozier has previously pointed out that "Floris" may come from [Floridsdorf](https://en.wikipedia.org/wiki/Floridsdorf), a district of the capital Vienna (Austria). May be possible that "Florids" got converted to "Floris" in my head (because I definitely have heard this name beforehand), but it was 100% not intended and I would have never thought of this association until it was pointed out to me.

So yeah that's the story behind the name :)

**Reply by Nikoichu** [reply]
I see, thanks for the backstory, this was super interesting to read :3
The association I'm making with "Floris" is that it's somehow tied to the word "Floral", eg planty greeny growy thingies.

I was asking, because I am considering making an app icon for you, because the current one is pretty generic and bland (common amongst open source apps unfortunately). I'm a graphic designer with a need for projects in my portfolio, so I thought, what the hell, why not contribute to an open source project that I like?

So, got any suggestions / recommendations / requests for a unique app icon? Text/initials on it? Perhaps some specific graphic elements, like the keyboard keys you have currently? Any colors or meaning you want the logo to convey?

**Reply by Glitchy-Tozier** [reply]
1. (I actually like the current icon so I don't feel the need to see it change)
2. If it _were_ to change, the first thing I'd think of is flowers & growing plants. Probably because of its [latin meaning](https://www.latin-is-simple.com/en/vocabulary/noun/9694/).

**Reply by Nikoichu** [reply]
It's fine if you don't want a new icon, I just thought I'd offer.
Here, I made some sketches, let me know if anything catches your eye. Keep in mind that anything is change-able, and elements from the different variants can be combined. Also they will look a lot cleaner once I vectorize them.

![Floris](https://user-images.githubusercontent.com/42444745/161562174-49c32682-51a0-4eb0-920a-d4cf9ad5119c.jpg)

**Reply by patrickgold** [reply]
Now that I think about it the "flower" association makes a lot of sense and it is an association with something positive.

I am a bit conflicted on this topic, because I both like the simple look of the current icon and want to keep it but also your sketches look very cool!

What definitely caught my eye is the flower in the bottom right, relatively simplistic but looks very cool. This ontop of a keyboard surface like the one in the left (or even stand-alone) could definitely look very unique. Important though is that the app icon leaves some space for the beta and debug words and that it is immediately noticeable without having to decipher small white letters (that's why the track names are color coded in the current icon design).

Personally I would not have anything against a new icon if it provides a unique identity to the app in the long term. @Glitchy-Tozier @tsiflimagas what do you both think about the sketches and the idea above?

Something I am completely trying to avoid though is abbreviating FlorisBoard with "FB", because this abbreviation is already heavily used for Facebook, and I don't want to have anything distant to do with that name.

**Reply by Glitchy-Tozier** [reply]
- I agree that the flower on the bottom-right looks nice.
- My favorite icon is the one on the top-left though. I like how it still shows the keyboard (which is probably similar to the old icon).
- The icon on the top-middle looks fun, but I think there's too much going on for it to be used as a small icon. {keyboard + it's tilted + flower + shadow}

General thoughts:
1. Those icons will be displayed in small sizes. I think it's a good idea to
    - keep details low
    - not add any text (except for maybe a color-coded beta-badge) (notice the alliterations :) ?)
    - keep used colors to a minimum (as you did pretty well. I still get the feeling that keyboard + flower might be overwhelming)
2. Personally I'm a fan of icons without unnecessary shadows

**Reply by Nikoichu** [reply]
Yeah, I don't like text in icons either, that's why I wrote "optional" on each of them. Don't think I'm gonna include any text in the designs, apart from "beta" and "debug". As soon as I wrote down the "FB" I knew it was a bad idea, so I agree there too. The text and banners are gonna create problems for adaptive icons too.

I'm gonna try to use the bottom-right flower on both of the top keyboard variants (straight and perspective) and see what works best. Also the shadows are gonna be very subtle, like in all material design icons, but I could very easily remove it if it's not to your taste.

I kept your current color scheme, because I like it, and because it's going to link the new icon to the old identity, so people can make the association.

Will post back here with variants :3

**Comment by Nikoichu** [comment]
![Designer_hG1MsU0PJa](https://user-images.githubusercontent.com/42444745/162474328-b686a8bb-1497-477d-aeab-909f9b21ffcb.png)

Here are some variants.
Top row: My preferred icon, with "BETA" and "DEBUG" versions.
Bottom row: Current icon, dark variant with round keys, and outlined variant with shadows and square keys

I started with the dark mode, similar to the current icon, but it wasn't working out as I imagined it, until I switched the colors (grey keys and white background), which I think works a lot better and looks cleaner. Also I tried using the flat tones in your current icon, but they looked a bit soulless and boring being flat, so I added a gradient to breathe some life into them. I can tone it down, or even intensify it even more, depending on what you want.
I omitted the 45 degree "material design" shadow from the sketches, because it would have looked dirty on the white background and you didn't seem that excited for it anyway. Also, the perspective keyboard doesn't work :D

Let me know what you think and which elements you like.

EDIT:
![Designer_SwpcHzNeDu](https://user-images.githubusercontent.com/42444745/162480076-deb16d85-9cc5-4485-a65f-8e274ba9309b.png)

A new variant without white contours, and a slightly updated old variant with underglow (to separate the lotus from the keys)

**Reply by patrickgold** [reply]
These icon proposals look very cool, greatly appreciate your efforts for them! I had a look at all the given variations and elements and am amazed how many different options you came up with! Down below I go in details what I think of those:

> I started with the dark mode, similar to the current icon, but it wasn't working out as I imagined it, until I switched the colors (grey keys and white background), which I think works a lot better and looks cleaner.

Do you think so? You just have one icon with dark colors but I really like the color mix with the dark grey background and white keys. This is especially true because the "default" color theme throughout the app and keyboard UI is also in dark grey+white, so I really like these color scheme a lot.

> Also I tried using the flat tones in your current icon, but they looked a bit soulless and boring being flat, so I added a gradient to breathe some life into them. I can tone it down, or even intensify it even more, depending on what you want.

In the flower the gradients look really cool and indeed bring life to the icon, I agree. But in the keys (bottom left+right) they look a bit weird, how would it look if those 2 where flat or only a light intensity gradient?

>  I omitted the 45 degree "material design" shadow from the sketches, because it would have looked dirty on the white background and you didn't seem that excited for it anyway. Also, the perspective keyboard doesn't work :D

I agree, the 45° shadow does not work at all and omitting is the best way for this shadow.

As for the separation of the flower and the keyboard, I think on a dark background it would work best if it was separated by an empty space like the one in bottom row icon no. 2. I also really like the separation of the different flower components in that icon, really gives the different shapes a clear and consistent contour.

The key shapes IMO are better if they aren't rounded too much, like the ones in the first row. Gives the icon the same key shapes as the "real" keyboard.

Also something I don't know if it is really possible to fix, but the icon foreground comes dangerously close to the rounded icon border. would it be possible to have a little bit more spacing?

**Reply by Nikoichu** [reply]
I'll try some more dark variants if it's your preferred color scheme. The trouble there is that the white keys kinda outshine the lotus icon, when it should be the other way around. Maybe I'll try making them light grey instead, will see.

Yeah, I was really wondering about the keycap gradients, will try a version without and a version with a toned down gradient.

So I take it you're not a fan of the "no-contours" version (first one after the EDIT)? I tried it out, because the white kinda chops the flower into pieces, whereas with a dark background, this might not be an issue, so this solution might not be needed.

It's no problem at all to scale the icon down a little to give it more breathing space. Note that the round icon shape I've chosen to work in is the most "restrictive" when it comes to space, so the icon might have more breathing room when placed in a square or squircle shape icon. You might need to test that in android studio when making the adaptive icon, since I don't have it installed on my PC. But we're getting ahead of ourselves.

**Reply by Nikoichu** [reply]
![Designer_8OCauVZkLL](https://user-images.githubusercontent.com/42444745/162492439-bc331e25-053d-4186-b9dc-ee9206ed124e.png)
Alright, those fixes were pretty quick and easy to implement. Anything else before I finalize and render it?

**Reply by Glitchy-Tozier** [reply]
1. I also prefer the dark variants
2. I feel like it's difficult to read that "beta"-text.

**Reply by Nikoichu** [reply]
Readability was also a concern for me, so I tested a 128x128 render:
![Florisboard_v3](https://user-images.githubusercontent.com/42444745/162508204-a0080c45-ac2e-4f0c-a3a7-9dbc343ca8cd.png) 
![Florisboard_v2](https://user-images.githubusercontent.com/42444745/162508212-752b53f5-a6e1-4f9c-a117-ba5ed6a11166.png) 
![Florisboard_v4](https://user-images.githubusercontent.com/42444745/162508228-6adf0814-0383-4d11-8d26-f946e5b66aa7.png)

128x128 is the size icons are displayed at on my phone (Poco X3 NFC), in the playstore they are around 154x154, but I can't speak for devices with smaller screens. Look at these icons at 100% zoom on your phones (load the desktop version of the page and outzoom at max, works on Firefox for android) and see if they are readable for you.

If you were referring to unreadability due to low tone/color contrast with the background, I could:
A) Lighten the green color a bit
B) Add a subtle glow/shadow that should increase readability
C) Darken the background further
(or multiple of the above)

**Reply by patrickgold** [reply]
I already really like the above renders, well done! I really like the flat look of the bottom keys, combines really great with the gradients of the flower.

Just a thing though: Is the spacebar a little bit higher than the other keys? Also the Beta and Debug font is different in size, and both sizes mismatch with the key heights if I see correctly.

As for the readability concern: The old Beta and Debug font also wasn't super-readable, important is that there's a noticeable difference between the tracks somewhere in the icon, in this case the space bar color. This is something which has to be checked on phone screens though, and may need little adjustments based on that.

**Reply by Glitchy-Tozier** [reply]
> If you were referring to unreadability due to low tone/color contrast with the background, ...

Exactly, that was my problem.

One additional idea: I like Mozilla's approach of giving the whole icon a different color-scheme (instead of using text). What do the icons look like if
- in the beta-release all of the flower is green
    - still keeping the color-gradients,
    - removing the bottom text or turning it grey
    - possibly turning the separator-lines within the flower white
- in the debug-release all of the flower is orange

**Reply by Nikoichu** [reply]
![Designer_XjDW7uMDI3](https://user-images.githubusercontent.com/42444745/162560667-44376c89-4f25-4a54-b940-e23a041c65eb.png)
Here, fixed the spacebar (nice catch) and the text (via the heretical method of stretching the letters :D), and lightened the green. I also darkened the grey and put a 5% blue in there, because "you should never use pure grey", which also seems to help with the contrast and overall color scheme.
Will try these alternate color variants when I get back home, but I don't know if they will work that well single-colored...
Another variant could be to just have a normal spacebar without text, and just have it colored differently for the different versions.

**Reply by Nikoichu** [reply]
Okay, here are a couple of variants for the BETA and DEBUG versions, take your pick off the flower buffet :D
![Designer_lDwJvxIT5x](https://user-images.githubusercontent.com/42444745/162568716-2e851205-98a0-45d8-88ba-c0991cebd01d.png)
The single-color icons work better than I thought, they really are reminiscent of Firefox's nightly icon.
EDIT: oops I swapped the text of the beta and debug icons on the 4th and 5th rows, but it's easily fixable.

**Reply by Glitchy-Tozier** [reply]
Very nice! :)
My favourites are those:
![image](https://user-images.githubusercontent.com/59611881/162574698-fc2cbb41-33a7-4cf1-913d-cd34fc545f07.png)
![image](https://user-images.githubusercontent.com/59611881/162574703-edfffb9a-2aec-4af5-a77b-52ddb8a32e36.png)

1. I think you might have switched the beta- & debug-colors. Beta = green, debug = orange.
2. The white beta & debug text that replaced the spacebar seems to be higher than the orange&green buttons besides it.

**Reply by Nikoichu** [reply]
@Glitchy-Tozier yeah I saw that I swapped them, I already fixed it in my file. Also, the text isn't actually higher, but it might visually seem so, so I might need to squish the text a little bit.
@patrickgold waiting on your feedback to finalize these and send them over to you guys <3

**Reply by patrickgold** [reply]
@BloodRaven0 really like these final proposals! You already stated that you locally swapped the colors, this is then fixed. As for the text: Could you do a side by side comparison of the text being in the flower color (beta green and debug orange) and white for both? I don't known, but I think the icons for those tracks will be even more recognizable if the text is in the separate color.

> Also, the text isn't actually higher, but it might visually seem so, so I might need to squish the text a little bit.

If you zoom in the text is indeed not higher, seems like our brains mess with our perception xD. Maybe you could make the text slightly less high, so for our perception it appears as the same height?

Apart from that I don't have any more feedback and think these are ready for integration testing :)

**Reply by Nikoichu** [reply]
Oops, I accidentally started a new comment chain.
...again.

**Comment by Nikoichu** [comment]
![Designer_PvvUql953d](https://user-images.githubusercontent.com/42444745/162585247-871d569a-ba6a-425e-bd19-e3ea6ff94968.png)
Here they are, colors fixed and text squished.
Personally I think the ones with the white text are better, because of the better readability, and better reminiscence of the main icon.
But as always, the call is yours, I'm just giving my opinion.

**Reply by patrickgold** [reply]
Thanks for the comparison! I just viewed the screenshot on my  phone and even though I still somewhat like the colored text more it is a lot less readable, so I think going for the white text is the best option. The text looks a lot better too, now that it got squished a bit.

I think these proposals can be seen as final, and we can move to testing it out by replacing the old icon in the debug version (after today's 0.3.15 release).

**Reply by Glitchy-Tozier** [reply]
I think the beta-text could still be squished, just by a few pixels though. It's not strictly necessary though.

> I think these proposals can be seen as final, and we can move to testing it out by replacing the old icon in the debug version (after today's 0.3.15 release).

I think (after testing it in debug-versions), it would be fun to introduce the new icon in a beta version.

**Reply by Nikoichu** [reply]
Okay, after reading a bunch of documentation on adaptive icons, I found out that it's better for them to be imported directly in vector format (SVG), and need to be sized at 108 dp (no idea what this stands for, turns out it's just pixels), but only the inner 72 dp get used for static icons, the rest are for animations. The inner 66 dp are a "safe zone" where the icon is guaranteed not to be cut off by different-shaped icon masks. And foreground and background need to be in separate layers (I presumed that meant separate files?)
Source: https://medium.com/androiddevelopers/vectordrawable-adaptive-icons-3fed3d3205b5

![Florisboard](https://user-images.githubusercontent.com/42444745/162606009-ad8a0057-a325-4b6c-bfea-df927a804fe2.svg)
![Florisboard_BETA](https://user-images.githubusercontent.com/42444745/162606014-75c493c1-8160-4e82-8101-267911438a51.svg)
![Florisboard_DEBUG](https://user-images.githubusercontent.com/42444745/162606016-843aa0ab-3467-4639-ae68-1a7c4fd56a8b.svg)
![Florisboard_Background](https://user-images.githubusercontent.com/42444745/162606096-196eca2e-8315-4f95-8219-dbd5a059c9ae.svg)
(oh wow I didn't know github had an SVG renderer... just "save as", they should still be in svg format)

Since the background is just a solid color, you might also be able to implement it as a hex code: 3D3E42
If you'd rather have PNGs, I can still render you some, but I don't know at what resolution they would be optimal. Also they'd probably be a bit bigger in filesize.

All copyrights to the final icons belong to you now. I only ask that I be allowed to include these in my design portfolio :3

Let me know if there is anything further I can do for you. Keep up the good work on this awesome keyboard, I'm eagerly awaiting the new updates! <3
P.S. I'm so uninstalling Gboard when words suggestions come out :D

**Reply by patrickgold** [reply]
Thanks again for your incredible work, really really appreciate it! The final renders look awesome and really a lot more unique than my generic keyboard icon before :)

> Okay, after reading a bunch of documentation on adaptive icons, I found out that it's better for them to be imported directly in vector format (SVG),

Yep adaptive icons must be in a vectorized format, do it can correctly render on all the different screen densities. Your given files are perfect. I do need some PNG files of this icon for store presence and such, but the Android Studio adaptive icon tool auto generates all the correct and different sizes automatically, so no need for a separate PNG render from your side.

> Since the background is just a solid color, you might also be able to implement it as a hex code: 3D3E42

Yup for adaptive icons a single hex color is defined and then used by the launchers and such as a background. What I just discovered rather by accident thanks to GitHub's SVG implementation: Your colors used in the flower look even more vibrant and lively if the icon background is more dark:

![image](https://user-images.githubusercontent.com/19412843/162612725-8ce6493a-ca37-4ecc-9b1f-b652f9a0c212.png)
![image](https://user-images.githubusercontent.com/19412843/162612740-ca4e13bb-aca8-4b95-b600-8fa9ed059e0e.png)
![image](https://user-images.githubusercontent.com/19412843/162612747-80a69fdc-f6ce-402c-ab2d-ba17a857966f.png)

This may be the only thing which I may experiment on when testing the icon in the debug version later today (as I just have to change the bg hex code). I fear though that a too dark background may lead to problems when the icon is amongst other apps icons and on app stores, but will see. I will check back on you later with screenshots how the icon looks when displayed as an adaptive icon.

> All copyrights to the final icons belong to you now. I only ask that I be allowed to include these in my design portfolio :3

Perfect, thanks a lot! Of course you are allowed to include above icons in your portfolio, it is your awesome work and time invested :) I will also make sure to include a credit note in the "Used components and icons" section of the README, if you want I can link your portfolio here as well!

> I think (after testing it in debug-versions), it would be fun to introduce the new icon in a beta version.

That's the plan :) 0.3.16-debug later today for experimenting, 0.3.16-beta01 in a few days after testing and then 0.3.16 stable.

**Reply by Nikoichu** [reply]
Glad you like them! <3
About the background, yeah a bit darker might look cooler, just don't go too dark, and test it on multiple screens, since each shows colors a bit differently, especially if it's not calibrated. Also my recommendation is not to use pure grey, add a little bit of color in there (about 5% blue works best imo).
About crediting, I would very much appreciate it. Here's my current portfolio link, you can just hyperlink it ontop of my name if you want: https://www.behance.net/nikolayanzarov

**Reply by patrickgold** [reply]
I now have added the icon to the app and experimented with the backgrounds. With adaptive icons on phone screens a dark background like I posted above does not look good at all, also not like somewhat darker than yours, so I stick with your background color (#3D3E42), which works incredibly well.

Some minor changes I had to do:
- Scaling of the foreground: 93% to stay clear of the danger zone of icons according to Android Studio and my own tests on the OnePlus launcher for instance
- I had to move the foreground slightly down, because else the foreground would not be centered in a squarish shape. This means the top within a circle appears to have slightly more margin than the bottom, this is due to the fact that the flower+keyboard is triangle-like. This aligns with the behavior of other app icons which are also a triangle-like shape.

Renders of the different shapes in Android Studio:
![image](https://user-images.githubusercontent.com/19412843/162624347-c946088c-3d42-441f-a86d-1bd2f12aafc9.png)
![image](https://user-images.githubusercontent.com/19412843/162624362-6bf72569-3085-4ef0-872b-146bc0d5b62c.png)
![image](https://user-images.githubusercontent.com/19412843/162624169-da022ce9-3ebd-42c1-a00f-1e9747b9bca5.png)

Comparison of the old stable and beta icon with the new icon on my homescreen:
![image](https://user-images.githubusercontent.com/19412843/162624678-804f6687-380d-45e1-b5b5-577939c57cd4.png)

See also PR #1775 implementing your icons. I also made sure that at least one of the commits is associated with your GitHub account so GitHub additionally picks you up as a contributor and adds you to the list of contributors.

What do you both think? If you need more screenshots I can provide them too.  
Now from my side I will do some testing and need to update the Fastlane icon images once we all agree on the final icon, but generally else I think it is ready for merge.

**Reply by Glitchy-Tozier** [reply]
It looks really good, but just in case one of you guys has some more energy to spend, I have two nitpicks. Both only came up when looking at @patrickgold's new images:

1. I feel like the flower overpowers the rest of the icon. Not sure how to fix it, but we could try slightly brightening the keys' color, (align it with the previous icon's key-color)
2. The spacebar-texts might be unnecessary. I don't know about you two, but I can't read it without squinting my eyes at the icons and bringing the phone closer to my face.
On first glance it just looks like a slightly transparent spacebar to me. Imo we could just use the regular, white spacebar.

**Reply by patrickgold** [reply]
So I adjusted the key color in the original svg and set it to fully white, that's what the previous icon had. Also I removed the text of the space bar, so it would look like this:

![image](https://user-images.githubusercontent.com/19412843/162632560-4c14ae45-b51d-4f18-83e2-1e4f7902339b.png)
![image](https://user-images.githubusercontent.com/19412843/162632574-be530f56-f9ba-4be0-888a-236fa852063a.png)
![image](https://user-images.githubusercontent.com/19412843/162632582-6823d4bb-fe25-40b0-a43d-40878b32a9af.png)

WHile it certainly looks cleaner now it is impossible to immediately tell which icon belongs to which track, without prior knowledge at least. So don't really know how we fix this.

**Reply by Nikoichu** [reply]
Nice, looking good! :3
Well, about the text, almost everywhere you see the icon, it's gonna be accompanied by the app name, so I don't think it's that big of a deal when it comes to differentiating between the three. Once a user gets used to the color-coding, it should be fine.
I made the keys 90% grey, because I felt that they were actually overpowering the flower, and I intended it to be the primary element, but pure white works too.
Cheers! <3

**Reply by patrickgold** [reply]
Yeah you are right, most of the time the app name is besides the icon anyways. I modified the PR #1775 now to reflect those changes.

@BloodRaven0 here are the final SVG files, so you have them too:

<details>
<summary>Base file for editing (Inkscape SVG)</summary>

![florisboard_logo_v4_fg](https://user-images.githubusercontent.com/19412843/162636173-e1ccc668-b90a-4454-94c2-f418f6b3bd35.svg)
</summary>
</details>

<details>
<summary>Final Icon Stable (Foreground only)</summary>

![florisboard_logo_v4_fg_stable](https://user-images.githubusercontent.com/19412843/162636194-280b679b-0451-4f49-b1e9-aea7510b2bb0.svg)
</summary>
</details>

<details>
<summary>Final Icon Beta (Foreground only)</summary>

![florisboard_logo_v4_fg_beta](https://user-images.githubusercontent.com/19412843/162636202-2922b46d-2a1c-49da-b670-80eed92f1f61.svg)
</summary>
</details>

<details>
<summary>Final Icon Debug (Foreground only)</summary>

![florisboard_logo_v4_fg_debug](https://user-images.githubusercontent.com/19412843/162636214-d17d24ea-28ac-4e57-8394-4128b753ab28.svg)
</summary>
</details>

Again thanks a lot for your efforts and awesome work, really appreciate it! Will still test out the changes on some other screens but do not see any problems anymore. After testing I will merge the icon in tomorrow.

**Reply by Glitchy-Tozier** [reply]
> I made the keys 90% grey, because I felt that they were actually overpowering the flower, and I intended it to be the primary element, but pure white works too.

That reasoning makes a lot of sense, too. You two decide, I guess :)

**Comment by Glitchy-Tozier** [comment]
Two small things:
1. @BloodRaven0 could you assist with #1785? I think all that is needed is a png.
2. This repo still is using the old icon:

![Screenshot_20220413-084839_GitHub.jpg](https://user-images.githubusercontent.com/59611881/163118247-346f12d4-05e8-4d64-9354-ab3a115059a0.jpg)

**Reply by patrickgold** [reply]
> This repo still is using the old icon:

That's not the repo icon but the organization icon, seems like I forgot to change it. Will change that when I am back home.

**Reply by patrickgold** [reply]
Updated the org icon, will take a few minutes to propagate the change everywhere.

**Reply by Glitchy-Tozier** [reply]
Nice :)
The new icon really is good-looking!
